components/consumers/elasticsearch/main.go

```diff
@@ -61,19 +64,24 @@ func main() {
 	flag.Parse()
 
 	if err := parseFlags(); err != nil {
+		slog.Error("could not parse incoming flags", slog.Any("error", err))
 		log.Fatal(err)
 	}
 
+	slog.Debug("connecting to elasticsearch")
 	es, err := getESClient()
 	if err != nil {
-		log.Fatal("could not contact remote Elasticsearch: ", err)
+		slog.Error("could not contact remote Elasticsearch: ", slog.Any("error", err))
+		log.Fatal(err)
```

Comment:
do we need both the logging and the log.Fatal invocation? won't the `log.Fatal` just print the error anyway?

Reply:
great question and something i've been meaning to ask!
I want a better way to exit, ideally all of those would be cobra/viper without a log.Fatal but with a return err, therefore I think the structured logging here makes sense given a general plan to remove the "fatal" log line on 75